evm-adapters/src/blocking_provider.rs

```diff
@@ -20,13 +21,43 @@ impl<M: Clone> Clone for BlockingProvider<M> {
 
 impl<M: Middleware> BlockingProvider<M> {
     pub fn new(provider: M) -> Self {
-        Self { provider, runtime: Runtime::new().unwrap() }
+        Self { provider: Arc::new(provider), runtime: Runtime::new().unwrap() }
     }
 
     fn block_on<F: std::future::Future>(&self, f: F) -> F::Output {
         self.runtime.block_on(f)
     }
 
+    pub fn get_account(
+        &self,
+        address: Address,
+        block: Option<BlockId>,
+    ) -> eyre::Result<(U256, U256, Bytes)>
+    where
+        M: 'static,
+    {
+        // We clone the provider each time so that we can tokio::spawn it. Without
+        // tokio::spawning, we'd get concurrent polling of the combined future, but the
+        // actual underlying requests would not be done in parallel, c.f. try_join's
+        // docs.
+        let f = async {
+            let provider = self.provider.clone();
+            let balance = tokio::spawn(async move { provider.get_balance(address, block).await });
+
+            let provider = self.provider.clone();
+            let nonce =
+                tokio::spawn(async move { provider.get_transaction_count(address, block).await });
+
+            let provider = self.provider.clone();
+            let code = tokio::spawn(async move { provider.get_code(address, block).await });
+
+            tokio::try_join!(balance, nonce, code)
+        };
+        let (balance, nonce, code) = self.block_on(f)?;
```

Comment:
this will probably not be noticeable and potentially worse due to spawning overhead, most of the time will be spent waiting on the responses, and that all arrive at the exact same time is probably unlikely.

Reply:
Good call, https://github.com/gakonst/dapptools-rs/pull/70/commits/1a8ac3247175fe914dc7c4a709b599f99cf052e7